slack_sdk/http_retry/builtin_async_handlers.py

```diff
@@ -0,0 +1,46 @@
+from typing import Optional, List
+
+from aiohttp import ServerDisconnectedError, ServerConnectionError, ClientOSError
+
+from slack_sdk.http_retry.interval_calculator import RetryIntervalCalculator
+from slack_sdk.http_retry.state import RetryState
+from slack_sdk.http_retry.request import HttpRequest
+from slack_sdk.http_retry.response import HttpResponse
+from slack_sdk.http_retry.handler import RetryHandler, default_interval_calculator
+
+
+class AsyncConnectionErrorRetryHandler(RetryHandler):
```

Comment:
Since this async implementation relies on the same base class that is shared with the sync implementation, and the [base `RetryHandler` class' `prepare_for_next_request` uses the built-in Python's `sleep` method](https://docs.aiohttp.org/en/1.3/client.html?highlight=sleep) - could this lead to a situation where we block the process even when using an async handler?

I am not very familiar with aiohttp, but it seems like it is based on the asyncio library which has its own async-friendly `sleep` implementation (or, at least, [this aiohttp document page implies that such an async sleep exists](https://docs.aiohttp.org/en/1.3/client.html?highlight=sleep) - search for `asyncio` on this page for the relevant section).

I am posing this question from a place of ignorance and a desire to learn so it is likely I am completely off. But asking dumb questions is helpful for me to learn 🤪

Reply:
@filmaj Ah, this is a great point! Yes, we should use `asyncio.sleep` instead here and I was aware of it. But somehow I forgot to override the method. We can have a base class `RetryHandler`, which uses asyncio'sleep method. All the methods in it will be async methods. I will update this part shortly.